feat(#82): implement sandbox workflows

bra1n2h opened the pull request:
## Description
This PR introduces features for both creating and removing sandbox environments. The sandbox creation functionality ensures that developers can deploy isolated testing environments, while the sandbox removal feature automates the cleanup of resources when they are no longer needed (e.g., after a pull request is closed). These functionalities are integrated with AWS CodePipeline and CodeBuild to streamline infrastructure management.

## Related Issue
This change addresses [Issue #82](https://github.com/VilnaCRM-Org/website/issues/82), which involves automating the lifecycle of sandbox environments.

## Motivation and Context
The addition of sandbox creation and removal features is necessary to:

  Provide developers with temporary, isolated environments for testing new features.
  Automate the cleanup of unused resources to reduce costs and improve project efficiency.
  Ensure consistent infrastructure provisioning and decommissioning workflows.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] I have performed a self-review of my code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING.md**](https://github.com/VilnaCRM-Org/frontend-ssr-template/blob/main/CONTRIBUTING.md) document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] You have only one commit (if not, squash them into one commit).


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **Documentation**
	- Added comprehensive documentation for GitHub Actions workflows in `.github/sandbox_workflows.md`
	- Detailed instructions on workflow configurations, secrets management, and authentication

- **Chores**
	- Updated GitHub Actions workflow dependencies and action versions across multiple files
	- Updated caching mechanisms for dependencies in various testing workflows
	- Updated deployment workflow configurations

- **New Features**
	- Introduced new workflows for sandbox creation and deletion
	- Added automated secret rotation and validation processes
	- Enhanced deployment workflow with updated configurations and AWS role management
	- Introduced a new workflow for website deployment triggered by pushes to the main branch
	- Added alt text for images in various components to improve accessibility
	- Improved testing coverage with new test cases for alt tags in image components
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

bra1n2h committed:
feat(#1): implement sandbox workflows

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h commented on .github/workflows/sandbox-deletion.yml:
```diff
@@ -0,0 +1,41 @@
+          pipeline-name: sandbox-deletion
+          access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          region: eu-central-1
```
@coderabbitai I didn't get this suggestion since we check here if PR closed or merged and then we trigger pipeline

bra1n2h replied:
@coderabbitai , I got your point, how should this file look like then?

bra1n2h committed:
refactor(#1): edit yml files

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

Kravalg reviewed (changes requested):
Coderabbit should approve the PR first, then you can request review from me

bra1n2h committed:
refactor(#2): edit variables and validation

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h committed:
refactor(#3): edit formatting and add validation

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h committed:
refactor(#4): edit format file

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

Kravalg commented on .github/workflows/sandbox-rebuild.yml:
```diff
@@ -0,0 +1,46 @@
+  AssumeRoleAndStartExecution:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
```
The first job should be to check if the current push to the main branch and if it’s the case, then we should stop to run this action, because we don’t need to have sandbox rebuild for the main branch

bra1n2h replied:
This file no longer exists.

Kravalg reviewed (changes requested):
You have three different methods how to run the AWS codepipeline in GitHub actions
Please use only one method for running codepipeline for all GitHub actions

dependabot[bot] committed:
build(deps): bump @mui/material from 6.1.6 to 6.1.7 (#81)

* build(deps): bump @mui/material from 6.1.6 to 6.1.7

Bumps [@mui/material](https://github.com/mui/material-ui/tree/HEAD/packages/mui-material) from 6.1.6 to 6.1.7.
- [Release notes](https://github.com/mui/material-ui/releases)
- [Changelog](https://github.com/mui/material-ui/blob/master/CHANGELOG.md)
- [Commits](https://github.com/mui/material-ui/commits/v6.1.7/packages/mui-material)

---
updated-dependencies:
- dependency-name: "@mui/material"
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

* build(#81): update dependencies

---------

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: igor-pereimyvovk <igor.pereimyvovk@gmail.com>
Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

Kravalg committed:
wensite diagram

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h commented:
Deployment has started for version: http://sandbox-test-82-implement-sandbox-workflows.s3-website.eu-central-1.amazonaws.com

bra1n2h commented:
Latest Version is ready: http://sandbox-test-82-implement-sandbox-workflows.s3-website.eu-central-1.amazonaws.com 
    This deployed website will be automatically deleted after 7 days
    To keep it active, please make a new commit to trigger a redeployment

bra1n2h committed:
refactor(#5): remove rebuild and add it to creation and using OIDC connection

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h committed:
docs(#1): add documentation how to add secrets for sandbox workflows

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h committed:
docs(#2): edit contetns table

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h commented:
The deployment process for the sandbox has started. 
Please wait until the latest version is ready

bra1n2h commented:
Latest Version is ready: http://sandbox-test-82-implement-sandbox-workflows.s3-website.eu-central-1.amazonaws.com 
    This deployed website will be automatically deleted after 7 days
    To keep it active, please make a new commit to trigger a redeployment

dependabot[bot] committed:
build(deps): bump cross-spawn from 7.0.3 to 7.0.6 in the npm_and_yarn group (#85)

* build(deps): bump cross-spawn in the npm_and_yarn group

Bumps the npm_and_yarn group with 1 update: [cross-spawn](https://github.com/moxystudio/node-cross-spawn).

Updates `cross-spawn` from 7.0.3 to 7.0.6
- [Changelog](https://github.com/moxystudio/node-cross-spawn/blob/master/CHANGELOG.md)
- [Commits](https://github.com/moxystudio/node-cross-spawn/compare/v7.0.3...v7.0.6)

---
updated-dependencies:
- dependency-name: cross-spawn
  dependency-type: indirect
  dependency-group: npm_and_yarn
...

Signed-off-by: dependabot[bot] <support@github.com>

* build(#85): update dependencies

* build(#40): update pnpm lock yaml

* fix(#40): resolve issues after major version updates

---------

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: igor-pereimyvovk <igor.pereimyvovk@gmail.com>
Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h commented:
The deployment process for the sandbox has started.
Please wait until the latest version is ready

bra1n2h commented:
Latest Version is ready: http://sandbox-test-82-implement-sandbox-workflows.s3-website.eu-central-1.amazonaws.com 
    This deployed website will be automatically deleted after 7 days
    To keep it active, please make a new commit to trigger a redeployment

bra1n2h commented:
The deployment process for the sandbox has started.
Please wait until the latest version is ready

bra1n2h committed:
refactor(#6): using OIDC

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h committed:
refactor(#7): version change

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h commented:
Latest Version is ready: http://sandbox-test-82-implement-sandbox-workflows.s3-website.eu-central-1.amazonaws.com 
    This deployed website will be automatically deleted after 7 days
    To keep it active, please make a new commit to trigger a redeployment

bra1n2h commented:
The deployment process for the sandbox has started.
Please wait until the latest version is ready

bra1n2h committed:
docs(#3): add information about secrets for website deploy

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h committed:
docs(#4): edit table of contents and description

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h committed:
docs(#5): add information into introduction about deploy

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h commented:
The deployment process for the sandbox has started.
Please wait until the latest version is ready

bra1n2h commented:
The deployment process for the sandbox has started.
Please wait until the latest version is ready

bra1n2h commented:
Latest Version is ready: http://sandbox-test-82-implement-sandbox-workflows.s3-website.eu-central-1.amazonaws.com 
    This deployed website will be automatically deleted after 7 days
    To keep it active, please make a new commit to trigger a redeployment

bra1n2h commented:
The deployment process for the sandbox has started.
Please wait until the latest version is ready

bra1n2h commented:
The deployment process for the sandbox has started.
Please wait until the latest version is ready

bra1n2h commented:
Latest Version is ready: http://sandbox-test-82-implement-sandbox-workflows.s3-website.eu-central-1.amazonaws.com 
    This deployed website will be automatically deleted after 7 days.
    To keep it active, please make a new commit to trigger a redeployment.

bra1n2h committed:
refactor(#8): add token rotation trigger before sandbox creation

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h committed:
refactor(#9): change role variable name

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h committed:
refactor(#10): correct 2 lines accordong comments

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h committed:
refactor(#11): edit github rotation trigger based on the environment

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h committed:
refactor(#12): remove duplication

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h committed:
docs(#6): add information about additional secrets

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h committed:
refactor(#13): correct variables quoting

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h committed:
docs(#7): correct format

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h committed:
docs(#8): correct format

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h committed:
docs(#9): correct table of contents

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h commented:
The deployment process for the sandbox has started.
Please wait until the latest version is ready

bra1n2h commented:
Latest Version is ready: http://sandbox-test-82-implement-sandbox-workflows.s3-website.eu-central-1.amazonaws.com 
    This deployed website will be automatically deleted after 7 days.
    To keep it active, please make a new commit to trigger a redeployment.

bra1n2h committed:
refactor(#14): add newline for testing purpose

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h committed:
refactor(#15): remove unnecessary line

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h committed:
refactor(#16): update version for actions/cache and upload-artifacts

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h committed:
refactor(#17): add required variables for pipeline exectuion

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h committed:
docs(#10): update variables and correct formatting

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h committed:
docs(#11): remove unnecessary sentence

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h commented:
The deployment process for the sandbox has started.
Please wait until the latest version is ready

bra1n2h commented:
Latest Version is ready: http://sandbox-test-82-implement-sandbox-workflows.s3-website.eu-central-1.amazonaws.com 
    This deployed website will be automatically deleted after 7 days.
    To keep it active, please make a new commit to trigger a redeployment.

bra1n2h commented:
The deployment process for the sandbox has started.
Please wait until the latest version is ready

bra1n2h commented:
The deployment process for the sandbox has started.
Please wait until the latest version is ready

bra1n2h commented:
The deployment process for the sandbox has started.
Please wait until the latest version is ready

bra1n2h commented:
The deployment process for the sandbox has started.
Please wait until the latest version is ready

bra1n2h commented:
Latest Version is ready: http://sandbox-test-82-implement-sandbox-workflows.s3-website.eu-central-1.amazonaws.com 
    This deployed website will be automatically deleted after 7 days.
    To keep it active, please make a new commit to trigger a redeployment.

bra1n2h commented:
The deployment process for the sandbox has started.
Please wait until the latest version is ready

bra1n2h commented:
Latest Version is ready: http://sandbox-test-82-implement-sandbox-workflows.s3-website.eu-central-1.amazonaws.com 
    This deployed website will be automatically deleted after 7 days.
    To keep it active, please make a new commit to trigger a redeployment.

bra1n2h committed:
refactor(#18): edit to retrive secret with dynamic name

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h committed:
docs(#12): remove information about token secret

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

Kravalg committed:
chore: update CONTRIBUTING.md

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

Kravalg committed:
chore: update CONTRIBUTING.md

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

Kravalg committed:
chore: update CONTRIBUTING.md

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

Kravalg committed:
chore: update CONTRIBUTING.md

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

Kravalg committed:
chore: update CONTRIBUTING.md

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h commented:
The deployment process for the sandbox has started.
Please wait until the latest version is ready

bra1n2h commented:
The deployment process for the sandbox has started.
Please wait until the latest version is ready

bra1n2h commented:
Latest Version is ready: http://sandbox-test-82-implement-sandbox-workflows.s3-website.eu-central-1.amazonaws.com 
    This deployed website will be automatically deleted after 7 days.
    To keep it active, please make a new commit to trigger a redeployment.

bra1n2h commented:
The deployment process for the sandbox has started.
Please wait until the latest version is ready

bra1n2h commented:
Latest Version is ready: http://sandbox-test-82-implement-sandbox-workflows.s3-website.eu-central-1.amazonaws.com 
    This deployed website will be automatically deleted after 7 days.
    To keep it active, please make a new commit to trigger a redeployment.

bra1n2h commented:
The deployment process for the sandbox has started.
Please wait until the latest version is ready

Kravalg committed:
docs: improve README.md

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

Kravalg committed:
docs: improve README.md

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h commented:
The deployment process for the sandbox has started.
Please wait until the latest version is ready

bra1n2h commented:
:x: Health check failed for the sandbox bucket.
  Please check the CodeBuild logs for detailed error information.

bra1n2h committed:
refactor(#19): change secret name

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

Vladomira committed:
chore(deps): update packages (#109)

* chore: delete line in CONTRIBUTING and update screenshots for visual tests

Related to Dependabot PR updating @next/env to 15.1.4.

* chore: update screenshots

* chore: add prescript for updating playwright screenshots

* chore: format localization.json and other files, change script in package.json from pretest:visual to test:visual:update

* chore: update snapshots

* Update Playwright.Dockerfile

* Update README.md

* Update README.md

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h committed:
refactor(#82): update secrets names

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h committed:
refactor(#82): remove prod rotation for testing

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h committed:
refactor(#82): remove statement from needs block

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h committed:
fix(#82): remove unnecessary output

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h committed:
fix(#82): update workflows

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h committed:
fix(#82): add branch to trigger website deploy

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h committed:
docs(#82): correct information about variables

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

dependabot[bot] committed:
build(deps-dev): bump @types/node from 22.10.0 to 22.10.7 (#110)

* build(deps-dev): bump @types/node from 22.10.0 to 22.10.7

Bumps [@types/node](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/node) from 22.10.0 to 22.10.7.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/node)

---
updated-dependencies:
- dependency-name: "@types/node"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

* chore: update snapshots

* chore: update dependencies

* test: improve mutation test coverage

* test: improve mutation test coverage

* chore: update snapshots after updating dependencies

---------

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Vladomira <juliavladomira@gmail.com>
Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h committed:
fix(#82): remove prod steps for testing

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h committed:
fix(#82): fix pipeline execution

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h committed:
fix(#82): update workflow to get PR number

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h committed:
fix(#82): add permission

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h committed:
refactor(#82): remove testing branch

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h committed:
docs(#82): add necessary description

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h committed:
fix(#82): using curl to retrive PR number

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h committed:
fix(#82): update variables

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h committed:
fix(#82): fix workflow execution

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h committed:
refactor(#82): remove letter to test pipeline

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h committed:
refactor(#82): add prod stage to sandbox creation

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h committed:
refactor(#82): deploy testing

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h committed:
refactor(#82): change from test to prod

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h committed:
fix(#82): fix prod secret rotation

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h committed:
fix(#82): change variable

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h committed:
fix(#82): edit format check for token

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h committed:
fix(#82): correct token retrive

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h committed:
refactor(#82): add prod env

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

dependabot[bot] committed:
chore(deps-dev): bump next from 15.1.5 to 15.1.6 (#115)

Bumps [next](https://github.com/vercel/next.js) from 15.1.5 to 15.1.6.
- [Release notes](https://github.com/vercel/next.js/releases)
- [Changelog](https://github.com/vercel/next.js/blob/canary/release.js)
- [Commits](https://github.com/vercel/next.js/compare/v15.1.5...v15.1.6)

---
updated-dependencies:
- dependency-name: next
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h committed:
refactor(#82): testing pipeline

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h committed:
refactor(#82): remove testing actions

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h committed:
refactor(#82): remove workflow

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h committed:
refactor(#82): add newline

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h committed:
fix(#82): trigger pipeline when PR is closed

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h committed:
fix(#82): remove role-session-name

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h committed:
refactor(#82): change to test account id

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h committed:
refactor(#82): update error handling

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h committed:
refactor(#82): improve naming

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h committed:
docs(#82): enhance information

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h commented on .github/workflows/sandbox-deleting.yml:
```diff
@@ -0,0 +1,64 @@
+        env:
+          GITHUB_TOKEN: ${{ secrets.PAT_WITH_REPO_PERMISSIONS }}
+
+      - name: trigger sandbox deletion pipeline
+        run: |
+          PR_NUMBER="${{ env.PR_NUMBER }}"
+          if ! aws codepipeline start-pipeline-execution \
+            --name "sandbox-deletion" \
+            --variables \
+              "name=BRANCH_NAME,value=${BRANCH_NAME}" \
+              "name=PR_NUMBER,value=${PR_NUMBER}" \
+              "name=IS_PULL_REQUEST,value=0"; then
+            echo "::error::Failed to start pipeline execution"
+            exit 1
+          fi
```
@coderabbitai the specified role doesn't have permission to get pipeline status

bra1n2h commented on .github/workflows/sandbox-deleting.yml:
```diff
@@ -0,0 +1,64 @@
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          role-to-assume: arn:aws:iam::${{vars.PROD_AWS_ACCOUNT_ID}}:role/sandbox-deletion-trigger-role
+          role-session-name: GitHub_to_AWS_via_FederatedOIDC
+          aws-region: ${{ env.AWS_REGION }}
+
```
@coderabbitai github has automatic retries to assume the role

bra1n2h commented on .github/workflows/sandbox-creating.yml:
```diff
@@ -0,0 +1,183 @@
+              "name=IS_PULL_REQUEST,value=1"; then
+            echo "::error::Failed to start pipeline execution"
+            exit 1
+          fi
```
@coderabbitai the job should start pipeline to create or redeploy bucket, so this change isn't acceptable

bra1n2h commented on .github/workflows/sandbox-deleting.yml:
```diff
@@ -0,0 +1,64 @@
+          echo "PR_NUMBER=$PR_NUMBER" >> $GITHUB_ENV
+        env:
+          GITHUB_TOKEN: ${{ secrets.PAT_WITH_REPO_PERMISSIONS }}
+
```
@coderabbitai tell me what permissions should have this token to retrieve rate limit

bra1n2h committed:
refactor(#82): improve error handling

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h commented on .github/workflows/sandbox-deleting.yml:
```diff
@@ -0,0 +1,72 @@
+      - name: Get PR Number
+        id: get_pr_number
+        run: |
+          RATE_LIMIT=$(curl -s -H "Authorization: Bearer $GITHUB_TOKEN" \
+            -H "Accept: application/vnd.github+json" \
+            "https://api.github.com/rate_limit")
+          REMAINING=$(echo "$RATE_LIMIT" | jq .rate.remaining)
+          if [ "$REMAINING" -lt 10 ]; then
+            echo "::warning::GitHub API rate limit is low: $REMAINING requests remaining"
+          fi
```
@coderabbitai I think it will be better to use existing token, that we use to retrieve PR_NUMBER for e.x.

bra1n2h committed:
refactor(#82): add validation for variable

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

Kravalg commented on .github/sandbox_workflows.md:
```diff
@@ -0,0 +1,260 @@
+# Documentation for GitHub Actions Pipelines: Sandbox Creation, Sandbox Deletion and Website Deploy
```
Document name is sandbox_workflows, but we have website deploy there
Please move website deploy to the separated doc

bra1n2h replied:
done

Kravalg reviewed (changes requested):
Please rename the name for action and jobs
From sandbox pipeline execution to sandbox
From validate-secrets-rotation to check tokens
From trigger-secrets-rotation to rotate tokens
From sandbox-pipeline-executionto deploy

From deploy website to website

bra1n2h committed:
refactor(#82): rename jobs

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h commented on .github/workflows/sandbox-creating.yml:
```diff
@@ -0,0 +1,183 @@
+              "name=IS_PULL_REQUEST,value=1"; then
+            echo "::error::Failed to start pipeline execution"
+            exit 1
+          fi
```
@coderabbitai we already have validation for PR_NUMBER

bra1n2h commented on .github/workflows/deploy.yml:
```diff
@@ -9,34 +9,23 @@ env:
         run: |
-          aws codepipeline start-pipeline-execution --name ${{ secrets.PROD_AWS_WEBSITE_CODEPIPELINE_NAME }}
\ No newline at end of file
+          aws codepipeline start-pipeline-execution --name ci-cd-website-prod-pipeline
```
@coderabbitai we are following least privilege so role doesn't have permissions for this action

bra1n2h committed:
refactor(#82): improve error handling

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h committed:
docs(#82): remove deploy website

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h committed:
docs(#82): correct docs

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h committed:
docs(#82): add doc for website deploy

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h commented on .github/website_deployment.md:
```diff
@@ -0,0 +1,75 @@
+
+- Ensure that the IAM role (website-deploy-trigger-role) is correctly configured with the necessary permissions to trigger the AWS CodePipeline.
+- The vars.PROD_AWS_ACCOUNT_ID variable should be set up in the repository to contain the appropriate AWS account ID for the production environment.
+- The ci-cd-website-prod-pipeline CodePipeline should be created and configured in the specified AWS region.
```
@coderabbitai i did everything check please one more time

bra1n2h replied:
@coderabbitai I found just one line without newline, but I have language specifications

bra1n2h committed:
docs(#82): add additional information

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h committed:
docs(#82): correct permission example

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

bra1n2h committed:
refactor(#82): rename 1 job

Signed-off-by: Mykyta Mozghovoi <n.mozgowoi2017@gmail.com>

Kravalg reviewed (dismissed):
Awesome job 🚀

bra1n2h commented on .github/website_deployment.md:
````diff
@@ -0,0 +1,128 @@
+    aws codepipeline start-pipeline-execution --name ci-cd-website-prod-pipeline
+```
+
+## Security Best Practices for Managing Repository
+
+1. Use Least Privilege for IAM Roles
+
+When creating IAM roles for workflows, always apply the principle of least privilege. Only grant the permissions necessary for the task. For example, only grant codepipeline:StartPipelineExecution and codepipeline:GetPipelineState permissions for triggering the pipeline, instead of full access to CodePipeline.
+
+Refer to the example IAM policy provided earlier to restrict access to only the required CodePipeline resource.
+
````
@coderabbitai I didn't get what do you want to see in this topic "Instructions for rotating AWS credentials", since we have only role as a credentials

bra1n2h committed:
docs(#82): correct information

bra1n2h commented on .github/sandbox_workflows.md:
```diff
@@ -0,0 +1,236 @@
+        sandbox-creating.yml for the Sandbox Creation workflow.
+        sandbox-deleting.yml for the Trigger Sandbox Deletion workflow.
+
+  Testing:
+    After setting up, test the workflows by opening and closing pull requests to verify that the pipelines are triggered correctly.
+    After making changes, test by opening a PR to ensure that the workflow checks and triggers rotation as expected, and then starts the pipeline execution.
+
```
@coderabbitai we don't have such test cases for sandboxes

Kravalg committed:
Merge branch 'main' into 82-implement-sandbox-workflows

Kravalg reviewed (approved):
Good job 🚀

wrightalexander654 commented:
45ae835
